sql.py

```diff
@@ -35,27 +36,38 @@ def __init__(self, session, clazz):
         self.session = session
         self.clazz = clazz
 
+    @contextmanager
+    def _session_op(self):
+        try:
+            yield self.session
```

Comment:
If I understand this correctly, you are using `yield` for it's "give an object and do stuff afterwards" semantic and not because session is a generator.  If so, that would be worth a comment.  Also consider whether or not you can get the same effect more clearly by using a `finally` block.

Reply:
While weird-looking, that's a pretty standard pattern when used with [the `contextlib.contextmanager` decorator](https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager), so I would think it might be okay here.